examples/nitclk/synchronize-tclk.py

```diff
@@ -0,0 +1,167 @@
+            session_name = "session" + str(i),
+            resource_name = resource,
+            option_string = ""
+        ))
+        vi = init_with_channels_resp.vi
+        sessions.append(vi)
+        CheckForError(nifgen_service, vi, init_with_channels_resp.status)
+
+        # Configure channels
+        config_channels_resp = nifgen_service.ConfigureChannels(nifgen_types.ConfigureChannelsRequest(
+            vi = vi,
+            channels = "0"
+        ))
+        CheckForError(nifgen_service, vi, config_channels_resp.status)
+
+        # Configure output mode
+        config_out_resp = nifgen_service.ConfigureOutputMode(nifgen_types.ConfigureOutputModeRequest(
+            vi = vi,
+            output_mode = 1
+        ))
+
+        # Configure sample rate
+        config_sample_rate_resp = nifgen_service.ConfigureSampleRate(nifgen_types.ConfigureSampleRateRequest(
+            vi = vi,
+            sample_rate = sample_rate
+        ))
+        CheckForError(nifgen_service, vi, config_sample_rate_resp.status)
+
+        # Create waveform
+        create_waveform_resp = nifgen_service.CreateWaveformF64(nifgen_types.CreateWaveformF64Request(
+            vi = vi,
+            channel_name = "0",
+            waveform_data_array = waveform_data
+        ))
+        CheckForError(nifgen_service, vi, create_waveform_resp.status)
+
+        i += 1
+
+    config_hom_trig_resp = nitclk_service.ConfigureForHomogeneousTriggers(nitclk_types.ConfigureForHomogeneousTriggersRequest(
+        sessions = sessions
+    ))
+    CheckForError(nitclk_service, None, config_hom_trig_resp.status)
+
+    # Synchronize and start generation
+    sync_resp = nitclk_service.Synchronize(nitclk_types.SynchronizeRequest(
+        sessions = sessions,
+        min_tclk_period = 0
+    ))
+    CheckForError(nitclk_service, None, sync_resp.status)
+    initiate_resp = nitclk_service.Initiate(nitclk_types.InitiateRequest(
+        sessions = sessions
+    ))
+    CheckForError(nitclk_service, None, initiate_resp.status)
+
+    print(f"Generating square wave with sample rate {sample_rate} on {resources}...")
+
+    # Plot waveform
+    fig = plt.gcf()
+    fig.canvas.manager.set_window_title('Sample Waveform')
+    plt.plot(waveform_data)
+    plt.suptitle("Close the window to stop generation", fontsize=10)
+    plt.xlabel("Samples")
+    plt.ylabel("Amplitude")
+    plt.show()
+
+# If NI-FGEN API throws an exception, print the error message  
+except grpc.RpcError as rpc_error:
+    error_message = rpc_error.details()
+    if rpc_error.code() == grpc.StatusCode.UNAVAILABLE :
+        error_message = f"Failed to connect to server on {server_address}"
+    elif rpc_error.code() == grpc.StatusCode.UNIMPLEMENTED:
+        error_message = "The operation is not implemented or is not supported/enabled in this service"
+    print(f"{error_message}") 
+finally:
+    if 'vi' in vars() and vi.id != 0:
```

Comment:
This needs to be updated to close all sessions we have opened. You will need to iterate over `sessions` and whichever of them have non-zero `id`, call `Close` on them.

Reply:
changed that